pkg/trace/agent/agent.go

```diff
@@ -179,6 +189,8 @@ func (a *Agent) Process(t pb.Trace) {
 		return
 	}
 
+	defer a.timer.Time("datadog.trace_agent.internal.process_trace_ms", time.Now())
```

Comment:
`time.Now())` will be evaluated when this defer is declared, not when the defer executes, so I think all of these defer calls need to be done inside of anonymous functions.

Reply:
Correct, the argument is executed here on this line. This is working as intended. `time.Now()` marks the _start_ time from which the end time which be calculated once the defer runs, after the function returns and the goroutine exits.

Reply:
Oh I thought this behaved the same way as the statsd library. But it actually expects this argument to be "since" 

This is ok then